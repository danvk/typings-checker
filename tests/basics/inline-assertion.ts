// This checks the types of parameters to a callback using inline assertions.

function mapObject<T, U>(
  o: T,
  callback: (val: T[keyof T], key: keyof T, collection: T) => U
): {[k in keyof T]: U} {
  return {} as any;
}

// $ExpectType { a: string; b: string; }
mapObject({a: 1, b: 2}, (
  val,  // $ExpectType number
  key,  // $ExpectType "a" | "b"
  c,  // $ExpectType { a: number; b: number; }
) => '' + val);
