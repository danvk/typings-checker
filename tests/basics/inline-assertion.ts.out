tests/basics/inline-assertion.ts: 4 / 4 checks passed.
